src/ModelContextProtocol/Auth/OAuthAuthorizationHelpers.cs

```diff
@@ -0,0 +1,268 @@
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using ModelContextProtocol.Auth.Types;
+using ModelContextProtocol.Utils.Json;
+
+namespace ModelContextProtocol.Auth;
+
+/// <summary>
+/// Provides helper methods for handling OAuth authorization.
+/// </summary>
+public static class OAuthAuthorizationHelpers
```

Comment:
This needs to be turned into an instance type with an interface, registered as a singleton so that its implementation can be replaced by more secure and/or custom ones.